### DISCUSSION TITLE
Some useful ideas about OpenUtau

### DISCUSSION THREAD

**Discussion by Feishou0** [stakira/OpenUtau#1247] [Ideas]
1、I hope to set the engine used for a single note in the 'Note Settings', as this can give one or more notes different engine characteristics.
2、Custom added emojis can be permanently retained in the emoji bar, so you don't have to add them every time.

**Comment by Nexys4t** [comment]
Ideas are posted in the Discord server. GitHub is for bug reports.

1- What do you mean on the engine? The resampler? In the parameter settings you can add more resamplers and choose the one you wish individually for each note.

2- Custom added _emojis_? Do you have any screenshot about wich emojis are you talking about? As far as I am aware there is not even an "emoji bar"...

**Reply by Feishou0** [reply]
Sorry，This is the result of my machine translation.
“emojis”It means'expressions'.

**Reply by Feishou0** [reply]
![picture2](https://github.com/user-attachments/assets/cea885a4-aa64-4b16-b14f-1cd30b498d3d)
![picture1](https://github.com/user-attachments/assets/90507bc7-de71-4526-b5d4-9c152f59db7c)
I opened the note settings window and didn't see any resamplers available for selection.And I have placed the resampler in the corresponding folder. Please tell me how to operate it. Thank you.

**Reply by Nexys4t** [reply]
It's not in the note properties but in the parameters/expressions (the thing below the pianoroll at the bottom, you can change those in the gear icon at the left corner of that section)

Sorry for late replies

**Reply by Feishou0** [reply]
> It's not in the note properties but in the parameters/expressions (the thing below the pianoroll at the bottom, you can change those in the gear icon at the left corner of that section)
> 
> Sorry for late replies

No problem. But the one at the bottom left is about changing expressions, not the resampler that I said can change a single note.For example, I want to change the resampler of a certain note to 'moresample'.I want the pronunciation effect of other resamplers, but“wordline-R”It doesn't satisfy me.

**Reply by Nexys4t** [reply]
Yeah! In the expressions editor, in the Engine you have to write the exact name of the resamplers you want, each one separated by a comma.

Then in the Engine expression it should show up and then you can change a note's resampler individually

**Reply by Feishou0** [reply]
> Yeah! In the expressions editor, in the Engine you have to write the exact name of the resamplers you want, each one separated by a comma.
> 
> Then in the Engine expression it should show up and then you can change a note's resampler individually

emmmmm，I followed your method and although ENG can switch to resampler, using Flag did not have any effect. Can you tell me specifically what to do? Thank you.

**Reply by Nexys4t** [reply]
> > Yeah! In the expressions editor, in the Engine you have to write the exact name of the resamplers you want, each one separated by a comma.
> > Then in the Engine expression it should show up and then you can change a note's resampler individually
> 
> emmmmm，I followed your method and although ENG can switch to resampler, using Flag did not have any effect. Can you tell me specifically what to do? Thank you.

I'm a bit late, sorry.
I'm not sure how Flags work in OU, but there's Expressions that you can set... I don't feel brave enough to say "do this" if I'm not confident in what I'm saying works or not. Sorry

**Reply by nagotown** [reply]
> Ideas are posted in the Discord server. GitHub is for bug reports.

it's fine to put ideas in discussions, bug reports go to issues. ideas/suggestions should also be allowed in issues, but it still hasn't been allowed.

**Comment by bekotoonz** [comment]
what about adding an potato dancer
